### DISCUSSION TITLE
Proposal: Attribute manager

### DISCUSSION THREAD

**Discussion by DennisKost** [dotnet/csharplang#6721] [Language Ideas]
problem: I can have thousands of attributes of the same name in the project
solution: add attribute manager for managing adding attribute (call one constructor of manager, and then manager append needed attributes using constructor arguments), it define attributes in one place

```
public interface IAttributeManager : Attribute
{
    public ICollection<Attribute> Attributes { get; }
}

public class AttrContainer : IAttributeManager
{
    public ICollection<Attribute> Attributes { get; }
    
    public AttrContainer(string str1, bool b2)
    {
        Attributes = new[]
        {
            new Attribute1(str1),
            new Attribute2(b2),
            new Attribute3()
        };
    }
}
```

before:
```
[Attribute1("str1")]
[Attribute2(false)]
[Attribute3]
public string Prop { get; }
```

after:
```
[AttrContainer("str1", false)]
public string Prop { get; }
```

**Comment by RikkiGibson** [comment]
If the attributed declaration is partial, you might be able to build something like this using source generators. (partial properties aren't here yet, but we're hoping to do it in an uncoming language version.)
